### PR TITLE
cargo-generate: 0.5.3 -> 0.11.0

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Tests disabled because they seem to require networking now.

**Comment by Ma27** [comment]
Are all tests affected by this? Otherwise it's perhaps preferable to disable just the problematic testcases :)

**Comment by matthiasbeyer** [comment]
> Are all tests affected by this? Otherwise it's perhaps preferable to disable
> just the problematic testcases :) 

just one, yep you're right. How do I do this properly?

File: pkgs/development/tools/rust/cargo-generate/default.nix [hunk 1]
```diff
@@ -2,27 +2,23 @@
...
 
   buildInputs = [ openssl  ]
     ++ lib.optionals stdenv.isDarwin [ Security libiconv curl ];
 
-  preCheck = ''
-    export HOME=$(mktemp -d) USER=nixbld
-    git config --global user.name Nixbld
-    git config --global user.email nixbld@localhost.localnet
-  '';
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-generate/default.nix:21]: Please add a comment why tests are disabled.

**Comment by SuperSandro2000** [comment]
> Are all tests affected by this? Otherwise it's perhaps preferable to disable just the problematic testcases :)
> just one, yep you're right. How do I do this properly?

See https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/text/mdcat/default.nix#L32 for an example

**Comment by dywedir** [comment]
Thanks! Update merged in https://github.com/NixOS/nixpkgs/pull/148020

**Pull request closed without merging**